clang/runtime/dpct-rt/include/dpct/bindless_images.hpp

```diff
@@ -117,6 +137,341 @@ class image_mem_wrapper {
   image_mem_wrapper *_sub_wrappers{nullptr};
 };
 
+#ifdef _WIN32
+std::unordered_set<ID3D11Resource *> uniqueD3D11Res;
+#endif
+
+/// The wrapper class of graphics interop memory handle.
+class interop_mem_wrapper {
```

Comment:
This class should be common to both Win and Lin platforms.

Only the constructor `interop_mem_wrapper(ID3D11Resource* d3d11_res, int reg_flags)` is available for Win.
And another constructor `interop_mem_wrapper(GLuint gl_res, int reg_flags)` to be added in the future for OpenGL support will be available for both platforms.

Similar logic applies for Vulkan and other Graphics libraries

Reply:
The initialization seems very different for the platforms. I suggest put the common logic in a base class, and the implementation for different platforms can be put in derived classes, or classes have a same name but separated by macros. I think it can be more readable.